src/core/thread/mlr_manager.cpp

```diff
@@ -447,7 +447,7 @@ void MlrManager::HandleMulticastListenerRegistrationResponse(Coap::Message
 
     uint8_t      status;
     Error        error;
-    Ip6::Address failedAddresses[Ip6AddressesTlv::kMaxAddresses];
+    Ip6::Address failedAddresses[Ip6AddressesTlv::kMaxAddresses]{};
```

Comment:
LGTM.

It may be better to use the `Array<Ip6::Address`> for this (maybe in a PR).